server/src/main/java/org/apache/druid/client/indexing/SamplerResponse.java

```diff
@@ -58,6 +64,27 @@ public int getNumRowsIndexed()
     return data;
   }
 
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SamplerResponse that = (SamplerResponse) o;
+    return getNumRowsRead() == that.getNumRowsRead() &&
+           getNumRowsIndexed() == that.getNumRowsIndexed() &&
+           Objects.equals(getData(), that.getData());
```

Comment:
this may not do the correct thing always.

Reply:
What case did you have in mind where this would break down?

Reply:
Never mind. I was thinking if, for some `List` implementations, the `equals` may not have been implemented correctly. But that is unlikely to happen.  was also wondering if we care about the order too.

Reply:
Thanks, IMO responses that have different orders should be considered non-equal